framework/src/modules/token/cc_commands/cc_transfer.ts

```diff
@@ -49,104 +46,140 @@ export class CCTransferCommand extends BaseCCCommand {
+						tokenID,
+						amount,
+						receivingChainID,
+					},
+					TokenEventResult.TOKEN_NOT_SUPPORTED,
+				);
+
+				throw new Error(`tokenID ${tokenID.toString('hex')} is not supported`);
+			}
+		} catch (error) {
+			if (!(error instanceof NotFoundError)) {
+				throw error;
+			}
 		}
 
-		if (
-			!tokenSupported(this._supportedTokenIDs, params.tokenID) &&
-			ccm.fee >= ctx.ccmSize * MIN_RETURN_FEE &&
-			ccm.status === CCM_STATUS_OK
-		) {
-			await this._interopMethod.error(methodContext, ccm, CCM_STATUS_TOKEN_NOT_SUPPORTED);
+		if (status !== CCM_STATUS_OK) {
+			recipientAddress = senderAddress;
+		}
+
+		const userStore = this.stores.get(UserStore);
+		const recipientExist = await userStore.has(
+			methodContext,
+			userStore.getKey(recipientAddress, tokenID),
+		);
+
+		if (!recipientExist) {
+			if (fee < FEE_CCM_INIT_USER_STORE) {
+				throw new Error('Insufficient fee to initialize user account.');
+			}
+
+			// should replace feeAddress with trsSender address
+			await this._tokenMethod.burn(methodContext, feeAddress, tokenID, amount);
 		}
 
-		// let { recipientAddress } = params;
-		// if (ccm.status !== CCM_STATUS_OK) {
-		// 	recipientAddress = params.senderAddress;
-		// }
-
-		// const canonicalTokenID = await this._tokenMethod.getCanonicalTokenID(
-		// 	methodContext,
-		// 	params.tokenID,
-		// );
-		// const recipientExist = await this._tokenMethod.accountExists(
-		// 	methodContext,
-		// 	params.recipientAddress,
-		// );
-		// let receivedAmount = params.amount;
-		// if (!recipientExist) {
-		// 	const minBalance = this._minBalances.find(mb => mb.tokenID.equals(canonicalTokenID))?.amount;
-		// 	if (!minBalance || minBalance > params.amount) {
-		// 		if (ccm.fee >= MIN_RETURN_FEE * ctx.ccmSize && ccm.status === CCM_STATUS_OK) {
-		// 			await this._interopMethod.error(methodContext, ccm, CCM_STATUS_MIN_BALANCE_NOT_REACHED);
-		// 		}
-		// 		return;
-		// 	}
-		// 	receivedAmount -= minBalance;
-		// 	const [canonicalChainID, canonicalLocalID] = splitTokenID(canonicalTokenID);
-		// 	if (canonicalChainID.equals(CHAIN_ID_ALIAS_NATIVE)) {
-		// 		const supplyStore = this.stores.get(SupplyStore);
-		// 		const supply = await supplyStore.get(methodContext, canonicalLocalID);
-		// 		supply.totalSupply -= minBalance;
-		// 		await supplyStore.set(methodContext, canonicalLocalID, supply);
-		// 	}
-		// }
-
-		// if (tokenChainID.equals(ownChainID)) {
-		// 	const escrowStore = this.stores.get(EscrowStore);
-		// 	const escrowKey = Buffer.concat([ccm.sendingChainID, tokenLocalID]);
-		// 	const escrowData = await escrowStore.get(methodContext, escrowKey);
-
-		// 	escrowData.amount -= params.amount;
-		// 	await escrowStore.set(methodContext, escrowKey, escrowData);
-		// }
-
-		// const userStore = this.stores.get(UserStore);
-		// await userStore.addAvailableBalanceWithCreate(
-		// 	methodContext,
-		// 	recipientAddress,
-		// 	canonicalTokenID,
-		// 	receivedAmount,
-		// );
+		try {
```

Comment:
i think this try-catch is also not needed

Reply:
Habit I have for wrapping async functionality in try...catch. Updating....